platform/ios/src/MGLCameraChange.h

```diff
@@ -0,0 +1,35 @@
+#import "MGLFoundation.h"
+
+typedef NS_OPTIONS(NSUInteger, MGLCameraChangeReason)
+{
+    /// No reason for the camera move is specified.
+    MGLCameraChangeReasonNone = 0,
+
+    /// Set when a public API that moves the camera is called. This may be set for some gestures
+    MGLCameraChangeReasonProgrammatic = 1 << 0,
+
+    /// User tapped the compass to reset to North
+    MGLCameraChangeReasonResetNorth = 1 << 1,// Tap on compass
```

Comment:
How important is it to distinguish between a tap on the compass and a rotation gesture? If the developer does want to make this distinction, couldn’t they install a gesture recognizer on the compass view?